src/main/java/de/refactoringBot/refactoring/supportedRefactorings/RemoveMethodParameter.java

```diff
@@ -122,341 +149,296 @@ public String performRefactoring(BotIssue issue, GitConfiguration gitConfig) thr
+		for (String javaFile : allJavaFiles) {
+			// parse a file
+			FileInputStream filepath = new FileInputStream(javaFile);
+			CompilationUnit compilationUnit = LexicalPreservingPrinter.setup(JavaParser.parse(filepath));
 
-		// Search all classes
-		for (ClassOrInterfaceDeclaration currentClass : classes) {
-			// Check if class already marked as done
-			if (!allRefactorings.getDoneClasses().contains(currentClass.resolve().getQualifiedName())) {
-				// Check all implements + extends classes/interfaces
-				NodeList<ClassOrInterfaceType> impl = currentClass.getImplementedTypes();
-				NodeList<ClassOrInterfaceType> ext = currentClass.getExtendedTypes();
-				// If class implements of one of the done classes
-				for (int i = 0; i < impl.size(); i++) {
-					if (allRefactorings.getDoneClasses().contains(impl.get(i).resolve().getQualifiedName())) {
-						allRefactorings.addToDoClass(currentClass.resolve().getQualifiedName());
-					}
+			List<MethodCallExpr> methodCalls = compilationUnit.findAll(MethodCallExpr.class);
+
+			for (MethodCallExpr methodCall : methodCalls) {
+
+				ResolvedMethodDeclaration calledMethod = null;
+				try {
+					calledMethod = methodCall.resolve();
+				} catch (Exception e) {
+					continue;
 				}
-				// If class extends of one of the done classes
-				for (int i = 0; i < ext.size(); i++) {
-					if (allRefactorings.getDoneClasses().contains(ext.get(i).resolve().getQualifiedName())) {
-						allRefactorings.addToDoClass(currentClass.resolve().getQualifiedName());
+
+				if (refactoring.getMethodSignatures().contains(calledMethod.getQualifiedSignature())) {
+					refactoring.addMethodCall(methodCall);
+
+					if (!refactoring.getJavaFiles().contains(javaFile)) {
+						refactoring.addJavaFile(javaFile);
 					}
 				}
 			}
-		}
 
-		// Recursively build super class tree
-		for (String javaFile : allJavaFiles) {
-			allRefactorings = getSuperTree(allRefactorings, allJavaFiles, javaFile);
 		}
 
-		return allRefactorings;
+		return refactoring;
 	}
 
 	/**
-	 * This method creates refactoring objects to rename a method inside a specific
-	 * class and to rename all method calls for that specific method inside the
-	 * specific class.
+	 * This method returns all root-folders of java files (like the src folder or
+	 * the src/main/java folder from maven projects)
 	 * 
-	 * @param allRefactorings
-	 * @param methodSignature
-	 * @param classSignature
-	 * @param javaFile
 	 * @param allJavaFiles
-	 * @return
+	 * @param repoFolder
+	 * @return javaRoots
 	 * @throws FileNotFoundException
 	 */
-	private ParserRefactoringCollection createRefactoringObjects(ParserRefactoringCollection allRefactorings,
-			String methodSignature, String classSignature, String javaFile, List<String> allJavaFiles)
-			throws FileNotFoundException {
+	public List<String> findJavaRoots(List<String> allJavaFiles, String repoFolder) throws FileNotFoundException {
```

Comment:
I placed the method inside the new RefactoringHelper-Class. Not sure where else to put it. It is tied to refactoring and dynamic since package names can change, so I can't save those to our Config-Files that we store in our DB.

Reply:
I also placed all methods there that are used in multiple refactorings to reduce duplicate code.